### DISCUSSION TITLE
Edge: Double Arrow (Bidirectional)

### DISCUSSION THREAD

**Discussion by CezarCobuz** [xyflow/xyflow#1190] [New Features]
Looking at [Edges examples](https://reactflow.dev/examples/edges/) I did not find any double arrow type with pointers at both ends. Is there any way in which I can use a custom edge of this type? Or is there any built-in bidirectional edge type that I can use? 

![Screenshot 2021-05-11 at 10 23 32](https://user-images.githubusercontent.com/36707637/117777671-7fb52e80-b245-11eb-970f-46f1a6567a1f.png)

NOT a solution: A workaround I am thinking of is adding custom styles to the source and target <Handle /> of the nodes that look like pointers and use simple (unpointed)line edges but that feels a bit counterintuitive

**Comment by harsohailB** [comment]
Here is a solution that I developed for my use case that may help you:

Define a custom arrow marker (this example points back to source node)
```html
<defs>
      <marker
          id="markerId"
          markerWidth="12.5"
          markerHeight="12.5"
          viewBox="-10 -10 20 20"
          orient="auto"
          refX="0"
          refY="0"
        >
          <polyline
            stroke="#bbb"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1"
            fill="#bbb"
            points="5,-4 0,0 5,4 5,-4"
          ></polyline>
     </marker>
</defs>
```

Add `markerStart` to your edge path
```html
<path
   ...
   markerStart="url(#markerId)"
   ...
/>
```

_Optional: `data` prop for edges can be used to toggle bi-directional edge_
```javascript
const markerStart = data.isBidirectional ? "url(#markerId)" : null;
// then in path element
markerStart={markerStart}
```

Result
![image](https://user-images.githubusercontent.com/47438886/124969232-af817800-dfe3-11eb-8d8f-d5c5abac9c30.png)

**Reply by CezarCobuz** [reply]
Thank you! Worked like a charm!

**Comment by moklick** [comment]
we will add a `markerStart` and `markerEnd` option for edges in v10!

**Reply by pgupta56** [reply]
Great !! Thanks!!
